src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,207 @@
+            } else if (command.charAt(8) != ' ') {
+                throw new DukeException("Deadline", 0);
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if it is an 'Event' command.
+     *
+     * @param command Takes in command line input to process.
+     * @return returns true if it is said command.
+     * @throws DukeException Either a syntax error or lack of description.
+     */
+    private boolean isEvent(String command) throws DukeException {
+        if (command.length() >= 5 && command.startsWith("event")) {
+            if (command.length() == 5) {
+                throw new DukeException("Event", 1);
+            } else if (command.charAt(5) != ' ') {
+                throw new DukeException("Event", 0);
+            } else {
+                return true;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if it is a 'Delete' command.
+     *
+     * @param command Takes in command line input to process.
+     * @return returns true if it is said command.
+     * @throws DukeException Syntax errors, spacing.
+     */
+    private boolean isDelete(String command) throws DukeException {
+        if (command.length() >= 6 && command.startsWith("delete")) {
+            if (command.length() == 6) {
+                throw new DukeException("The Delete command must be followed by a number!");
+            } else {
+                if (command.charAt(6) != ' ') {
+                    throw new DukeException("The Delete command requires a number, separated by whitespace!");
+                } else if (isNumeric(command.substring(7))) {
+                    taskNumber = Integer.parseInt(command.substring(7));
+                    return true;
+                } else {
+                    throw new DukeException("The Delete command requires a number, separated by whitespace!");
+                }
+            }
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isFind(String command) throws DukeException {
+        int commandLength = command.length();
+        if (commandLength >= 4 && command.startsWith("find")) {
+            if (commandLength == 4) {
+                throw new DukeException("The Find command must be followed by a string!");
+            } else if (command.charAt(4) != ' ') {
+                throw new DukeException("The Find command requires a string, separated by whitespace!");
+            } else {
+                query = command.substring(5).trim();
+                return true;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    // Checks if String is numeric
+    private static boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+}
```

Comment:
Perhaps you could abstract a common private method to handle the different type of tasks? They seem like repeat code. The abstracted method can take in input of length of the command(Integer), name("deadline", event")

Reply:
Agreed, will include that in the future when I review my code. Thanks